source/Irrlicht/CBAWFile.cpp

```diff
@@ -271,4 +272,23 @@ size_t FinalBoneHierarchyBlobV0::calcNonInterpolatedAnimsByteSize() const
 	return keyframeCount * boneCount * scene::CFinalBoneHierarchy::getSizeOfSingleAnimationData();
 }
 
+bool runAes128gcm(const void* _input, size_t _inSize, void* _output, size_t _outSize, const unsigned char* _key, const unsigned char* _iv, bool _encrypt)
+{
+	const EVP_CIPHER* cipherType = EVP_aes_128_gcm();
+	EVP_CIPHER_CTX* ctx = EVP_CIPHER_CTX_new();
+
+	if (!ctx)
+		return false;
+
+	EVP_CipherInit_ex(ctx, cipherType, NULL, _key, _iv, _encrypt);
```

Comment:
i wonder if there is a way to copy a freshly intialized context so we don't have to init new ones all the time?
does EVP have a clone/copy function?